### ISSUE TITLE
AWS Transcribe to S3 bucket does not work with Cognito Unauth Credentials

### ISSUE THREAD

**Issue by ezhou7** [aws/aws-cli#4638]
Hi there,

I've come across an issue where AWS Transcribe is unable to use temporary security credentials when outputting speech-to-text results to an S3 bucket.  All it returns is an internal error with no message explaining anything.  These temporary security credentials are retrieved from a Cognito identity IAM role for unauthenicated users.  However, when I use an IAM user credentials, outputting to an S3 bucket works without any issues.  This issue appears in multiple places; I've run into this issue with AWS CLI, Swift and Python Boto3, all trying to make Transcribe output to an S3 bucket.

What does work without issue (for both the Cognito auto-generated IAM role and manually created IAM user) is running Transcribe without outputting to an S3 bucket.  This makes me think this has something to do with permissions to write to S3.

Here is the AWS CLI command:
```
aws transcribe start-transcription-job --transcription-job-name 2 --language-code en-US --media-format mp4 --media MediaFileUri=https://<my-input-bucket>.s3.us-east-2.amazonaws.com/<my-audio-file>.mp4 --output-bucket-name <my-output-bucket>
```
and this is the result with temporary security credentials
(aka. AWS CLI needs access key, secret key and a session token):
```
An error occurred (InternalFailureException) when calling the StartTranscriptionJob operation (reached max retries: 4): An internal error occurred.
```

Here is the Python code:
```
import boto3

def transcribe_service():
    transcribe = boto3.client(
        service_name="transcribe",
        region_name="us-east-2",
        aws_access_key_id="<access-key>",
        aws_secret_access_key="<secret-access-key>",
        aws_session_token="<session-token>"
    )
    response = transcribe.start_transcription_job(
        TranscriptionJobName="A56D8D80-5592-4904-9387-3E2D17B914BD",
        LanguageCode="en-US",
        MediaFormat="mp4",
        Media={
            "MediaFileUri": "https://<my-input-bucket>.s3.us-east-2.amazonaws.com/A56D8D80-5592-4904-9387-3E2D17B914BD.mp4"
        },
        OutputBucketName="<my-output-bucket>"
    )

    print(response)
```
and the result is the same as above:
```
botocore.errorfactory.InternalFailureException: An error occurred (InternalFailureException) when calling the StartTranscriptionJob operation (reached max retries: 4): An internal error occurred.
```

Here is the Swift code (if anyone is curious I use this library to call AWS in Swift - https://github.com/swift-aws/aws-sdk-swift):
```
class Transcriber : NSObject {
    var transcribeServiceClient: TranscribeService!
    
    override init() {
        super.init()

        let credentials = IdentityProvider().awsCredentials()
        transcribeServiceClient = TranscribeService(
            accessKeyId: credentials!.accessKeyId,
            secretAccessKey: credentials!.secretKey,
            sessionToken: credentials!.sessionToken,
            region: .useast2
        )
    }
    
    func transcribe(jobId: String) -> String {
        let mediaUriPrefix = "https://<my-input-bucket>.s3.us-east-2.amazonaws.com/"
        do {
            let media = TranscribeService.Media(mediaFileUri: mediaUriPrefix + jobId + ".mp4")
            let request = TranscribeService.StartTranscriptionJobRequest(
                languageCode: .enUs,
                media: media,
                mediaFormat: .mp4,
                outputBucketName: "<my-output-bucket>",
                transcriptionJobName: jobId
            )
            let future = transcribeServiceClient.startTranscriptionJob(request)
            let response = try future.wait()
            print(response)
            let mediaUri = response.transcriptionJob!.media!.mediaFileUri!

            return mediaUri
        } catch {
            print("Transcriber error: \(error)")
            print("Error message: \(error.localizedDescription)")
        }
        
        return ""
    }
}
```
and the result returns the same error:
```
Transcriber error: InternalFailureException: An internal error occurred.
Error message: The operation couldn’t be completed. (TranscribeService.TranscribeServiceErrorType error 2.)
```

I'm not sure if this has something to do with whether I am incorrectly setting the permissions or if there is something wrong with AWS Transcribe internally, and I'm stuck on this (haven't found any resources online that help me with this yet).

I've checked the AWS Transcribe blogs, but all I've found is this: https://aws.amazon.com/blogs/machine-learning/amazon-transcribe-now-lets-you-designate-your-own-amazon-s3-buckets-to-store-transcription-outputs/, which is not particularly helpful.

I've also found this: https://aws.amazon.com/getting-started/tutorials/create-audio-transcript-transcribe/, but the tutorial doesn't address this issue in any way.

Thanks in advance!

**Comment by kdaily** [comment]
Hi @ezhou7,

I apologize for the long delay in responding. Is this still an issue for you? Given that you have noticed it in AWS CLI, boto3 (which the AWS CLI uses) as well as AWS Swift, this is likely a limitation of the AWS Transcribe service. I agree that the error you're getting does not aid in debugging.

If you are still encountering it, I can raise it with the AWS Transcribe team. Thanks!

**Comment by ezhou7** [comment]
Hi @kdaily,

No worries for the delay. I'm not sure whether this is still an issue since I've moved from using the Transcribe S3 bucket service to the Transcribe real-time service some time ago. I no longer have the same setup as I did back then, so it's hard for me to test it again now. As for a workaround for this issue though, I ended up creating a user IAM key and using that instead.

Thanks,
Ethan
